realworld-api/src/main/java/com/study/realworld/user/controller/UserController.java

```diff
@@ -0,0 +1,56 @@
+package com.study.realworld.user.controller;
+
+import static org.springframework.http.ResponseEntity.ok;
+
+import javax.validation.Valid;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import com.study.realworld.user.controller.dto.request.JoinDto;
+import com.study.realworld.user.controller.dto.request.LoginDto;
+import com.study.realworld.user.controller.dto.request.UpdateDto;
+import com.study.realworld.user.controller.dto.response.UserInfo;
+import com.study.realworld.user.entity.User;
+import com.study.realworld.user.service.UserService;
+
+@Slf4j
+@RequiredArgsConstructor
+@RequestMapping(value = "/api")
+@RestController
+public class UserController {
+
+    private final UserService userService;
+    private final PasswordEncoder passwordEncoder;
+
+    @ApiOperation("SignIn")
+    @PostMapping("/users/login")
+    public ResponseEntity login(@RequestBody @Valid final LoginDto loginDto) {
```

Comment:
```suggestion
    public ResponseEntity<UserInfo> login(@RequestBody @Valid final LoginDto loginDto) {
```
제네릭 타입 대신 로 타입으로 반환하는건 조금 위험할 수 있어요   
로타입으로 설정할 경우 제네릭 문법을 받지 않기에 컴파일 타임에서 에러를 확인하기 어려워요   
예를 들면 return시 `ok(new String("abc"))`를 작성해도 런타임시에만 에러를 확인할 수 있게 됩니다.   
   
코드를 보면 일부러 사용하신 것 같은데 `ResponseEntity<?>`나 위에 제안으로 드린 코드도 고려해보시면 좋을 것 같아요

Reply:
[이펙티브 자바 item 26](https://catsbi.oopy.io/83635cfe-1cab-43f2-a943-56a9efd83fb2)을 참고하시면 좋을것 같아요 :)

Reply:
진영님 좋은 글 감사드립니다 !
로 타입,,, 제네릭,,, 메모...

Reply:
로직이 복잡하지도 않고, 다양한 리턴 형태도 존재하지 않아서, `ResponseEntity<T>.ok(T body)`로 들어가는 타입만을 고려하여 리턴하면 된다고 생각해서, 굳이 `ResponseEntity<UserInfo>` 방식으로 적지 않았었어요.

그러나 설명해주신 것처럼 로타입은 편리하지만, 분명 Smell이 나는 코드이기는 합니다..! 
반영하겠습니다.